sympy/stats/crv.py

```diff
@@ -422,7 +422,7 @@ def expectation(self, expr, var, evaluate=True, **kwargs):
                 p = poly(expr, var)
                 t = Dummy('t', real=True)
                 mgf = self._moment_generating_function(t)
-                if mgf is None:
+                if mgf is None or p.is_zero:
                     return integrate(expr * self.pdf(var), (var, self.set), **kwargs)
```

Comment:
If `p` is zero then that implies that `expr` is zero so is there any need to call `integrate` in that case?

Can we not just return zero?

Reply:
That'll be better. I'll make the necessary change.